src/main/java/com/minecolonies/coremod/colony/crafting/RecipeStorageFactory.java

```diff
@@ -207,6 +219,8 @@ public void serialize(IFactoryController controller, RecipeStorage input, Packet
         packetBuffer.writeInt(input.getSecondaryOutputs().size());
         input.getSecondaryOutputs().forEach(stack -> packetBuffer.writeItemStack(stack));
 
+        packetBuffer.writeResourceLocation(input.getLootTable());
```

Comment:
won't this crash if null?

Reply:
No, I don't think so. If so, I wouldn't be able to view any lists of recipes, and that works fine.

Reply:
I think I still prefer a null check here

Reply:
But we can't on the read, since this is a stream. By writing the null, the headers get written for the read to properly handle the null.

Reply:
What I always do is, write a boolean of "false" if null and "true" if not, and if not, we read the resLoc

Reply:
I thought that was encapsulated in the writeResourceLocation already?

Reply:
Looking it up, its converts to string, and THAT is what could fail but a null string does get handled correctly since it starts with a length byte.

Reply:
Null check in place.